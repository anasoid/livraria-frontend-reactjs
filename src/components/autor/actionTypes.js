export const GET_AUTORES = 'GET_AUTORES';
export const ADD_AUTOR = 'ADD_AUTOR';
export const REMOVE_AUTOR = 'REMOVE_AUTOR';