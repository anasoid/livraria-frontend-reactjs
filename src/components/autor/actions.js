import Helper from './helper';
import {GET_AUTORES} from './types';

export const getAutoresList = () => {
  const helper = new Helper();
  return dispatch => {
    return helper.getAutoresList()
      .then((data) => {
        return dispatch({type: GET_AUTORES, payload: data});
      }).catch();
  };
};

