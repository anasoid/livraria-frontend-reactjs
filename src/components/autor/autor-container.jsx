import React from 'react';
import AutorTable from './autor-table';

const AutorContainer = (props) => {
  return(
    <div>
      <h1>Novo Autor</h1>
      <form>
        <label htmlFor={'autorNome'}>Nome: </label>
        <input id={'autorNome'} name={'nome'}/>
        <br/>
        <label htmlFor={'autorEmail'}>Email: </label>
        <input id={'autorEmail'} name={'email'}/>
        <br/>
        <button id={'btnGravarAutor'} type={'submit'}>Gravar Autor</button>
      </form>
      <br/>
      <AutorTable/>
    </div>
  );
};

export default AutorContainer;