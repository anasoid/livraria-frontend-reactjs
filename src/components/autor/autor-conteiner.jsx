import React from 'react';

const AutorContainer = () => {
  return(
    <div>
      <h1>Novo Autor</h1>
      <form>
        <label htmlFor={'autorNome'}>Nome: </label>
        <input id={'autorNome'}/>
        <br/>
        <label htmlFor={'autorEmail'}>Email: </label>
        <input id={'autorEmail'}/>
        <br/>
        <button id={'btnGravarAutor'} type={'submit'}>Gravar Autor</button>
      </form>
      <br/>
      <table>
        <caption>Autores</caption>
        <thead>
          <tr>
            <th>Nome</th>
            <th>Email</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>Fulano</td>
            <td>fulano@admin.com</td>
            <td><button>alterar</button></td>
            <td><button>remover</button></td>
          </tr>
        </tbody>
      </table>
    </div>
  );
};

export default AutorContainer;