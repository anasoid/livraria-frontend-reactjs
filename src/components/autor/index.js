import React from 'react';
import AutorContainer from './autor-container';

const Autor = () => {
  return (
    <AutorContainer/>
  );
};

export default Autor;