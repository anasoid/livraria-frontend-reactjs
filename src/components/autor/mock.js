export const autores = {
  "autores": [
    {'nome': 'fulano', 'email': 'fulano@fulano.com'},
    {'nome': 'ciclano', 'email': 'ciclano@ciclano.com'}
  ]
};