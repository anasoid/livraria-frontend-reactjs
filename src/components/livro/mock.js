export const livros = {
  livros: [
    {titulo: 'reactjs', isbn: '12345678', preco: 20.0, dtLancamento: '11/05/2018', autores: [{nome: 'fulano', email: 'fulano@email.com'}]},
    {titulo: 'react native', isbn: '87654321', preco: 15.0, dtLancamento: '20/08/2018', autores: [{nome: 'ciclano', email: 'ciclano@email.com'}]}
  ]
}