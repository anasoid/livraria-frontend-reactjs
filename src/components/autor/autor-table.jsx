import React, {Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {getAutoresList} from './actions';

class AutorTable extends Component {

  componentWillMount() {
    this.props.getAutoresList();
  }

  render() {
    return (
      <div>
        <table>
          <caption>Autores</caption>
          <thead>
          <tr>
            <th>Nome</th>
            <th>Email</th>
          </tr>
          </thead>
          <tbody>
          {
            this.props.autores.map((autor, index) => {
              return (
                <tr key={index}>
                  <td>{autor.nome}</td>
                  <td>{autor.email}</td>
                  <td><button>alterar</button></td>
                  <td><button>remover</button></td>
                </tr>
              );
            })
          }
          </tbody>
        </table>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  autores: state.autor.autores
});

const mapDispatchToProps = dispatch => bindActionCreators({ getAutoresList }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(AutorTable);