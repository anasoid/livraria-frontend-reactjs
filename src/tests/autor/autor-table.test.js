import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import {AutorTable} from '../../components/autor/autor-table';
import {autores} from '../../components/autor/mock';

const getAutoresListMock = jest.fn();
const removeAutorMock = jest.fn();
const selectAutorMock = jest.fn();
const component = (
  <AutorTable autores={autores.autores}
    getAutoresList={getAutoresListMock}
    removeAutor={removeAutorMock}
    selectAutor={selectAutorMock}
  />
);

describe('test AutorTable components', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = mount(component);
  });

  it('should renders without crashing', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
    expect(getAutoresListMock).toBeCalled();
  });

  it('should be contain a table with title "Autores"', () => {
    const table = wrapper.find('table').at(0);
    expect(table.find('caption')).toHaveText('Autores');
  });

  it('should be contain a table with columns Nome, Email', () => {
    const rowsTh = wrapper.find('table').find('thead').find('tr').find('th');
    expect(rowsTh.at(0)).toHaveText('Nome');
    expect(rowsTh.at(1)).toHaveText('Email');
  });

  it('should be call props removeAutor when click button remover', () => {
    const rowsTBody = wrapper.find('table').find('tbody').find('tr').find('td');
    const removeButton = rowsTBody.find('button').at(1);

    removeButton.simulate('click');

    expect(removeAutorMock).toBeCalled();
  });

  it('should be call props selectAutor when click button alterar', () => {
    const rowsTBody = wrapper.find('table').find('tbody').find('tr').find('td');
    const alterarButton = rowsTBody.find('button').at(2);

    alterarButton.simulate('click');

    expect(selectAutorMock).toBeCalled();
  });

});
