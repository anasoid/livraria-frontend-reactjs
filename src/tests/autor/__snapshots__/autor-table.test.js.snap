// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test AutorTable components should renders without crashing 1`] = `
<div>
  <table>
    <caption>
      Autores
    </caption>
    <thead>
      <tr>
        <th>
          Nome
        </th>
        <th>
          Email
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          fulano
        </td>
        <td>
          fulano@fulano.com
        </td>
        <td>
          <button
            onClick={[Function]}
          >
            alterar
          </button>
        </td>
        <td>
          <button
            onClick={[Function]}
          >
            remover
          </button>
        </td>
      </tr>
      <tr>
        <td>
          ciclano
        </td>
        <td>
          ciclano@ciclano.com
        </td>
        <td>
          <button
            onClick={[Function]}
          >
            alterar
          </button>
        </td>
        <td>
          <button
            onClick={[Function]}
          >
            remover
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
